jupyterhub/app.py

```diff
@@ -1065,6 +1068,16 @@ def init_services(self):
                     raise AttributeError("No such service field: %s" % key)
                 setattr(service, key, value)
 
+            if service.managed:
```

Comment:
> This is the case where the OAuth is a service managed by the hub. 

OAuth isn't a service. The service would be something like nbviewer, formgrader, etc.

> Essentially JupyterHub acts as the OAuth provider and issues tokens.

Yes, JupyterHub itself is now an OAuth provider (always). Services may use OAuth to authenticate users.

This token block is unchanged, I just hand to move it up so that the API token is defined before the next block, which uses it.